packages/kbn-text-based-editor/src/ecs_metadata_helper.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { ESQLRealField } from '@kbn/esql-validation-autocomplete';
+import type { DatatableColumnType } from '@kbn/expressions-plugin/common';
+import type { FieldsMetadataPublicStart } from '@kbn/fields-metadata-plugin/public';
+
+const ECS_VERSION_FIELD = 'ecs.version';
+
+const removeKeywordSuffix = (name: string) => {
+  return name.endsWith('.keyword') ? name.slice(0, -8) : name;
+};
+
+/**
+ * Returns columns with the metadata/description (e.g ECS info)
+ * if available
+ *
+ * @param columns
+ * @param fieldsMetadata
+ * @returns
+ */
+export async function getColumnsWithMetadata(
+  columns: Array<{
+    name: string;
+    type: DatatableColumnType;
+  }>,
+  fieldsMetadata?: FieldsMetadataPublicStart
+): Promise<ESQLRealField[]> {
+  if (!fieldsMetadata || !columns.find((c) => c.name === ECS_VERSION_FIELD)) {
+    return columns;
+  }
+
+  try {
+    const fieldsMetadataClient = await fieldsMetadata?.getClient();
+    if (fieldsMetadataClient) {
+      const fields = await fieldsMetadataClient.find({
+        fieldNames: columns.map((c) => c.name),
+      });
+
+      if (fields.fields) {
+        return columns.map((c) => ({
+          ...c,
+          // Metadata services gives description for
+          // 'ecs.version' but not 'ecs.version.keyword'
+          // but both should show description if available
+          metadata: fields.fields[removeKeywordSuffix(c.name)],
+        }));
+      }
```

Comment:
Great usage of the FieldsMetadata service @qn895 👏 

> [!TIP]
> A further optimization here is to reduce a lot the API response payload by only keeping the field attributes you need, which I think it only consists of `description` so far! You can achieve it by passing the `attributes` option, this will return only the essential attributes for this use case and reduce quite a lot the API response size:
```ts
const {fields} = await fieldsMetadataClient.find({
  fieldNames: columns.map((c) => c.name),
  attributes: ['description']
});

if (fields) {
  return columns.map((c) => ({
    ...c,
    // Metadata services gives description for
    // 'ecs.version' but not 'ecs.version.keyword'
    // but both should show description if available
    metadata: fields[removeKeywordSuffix(c.name)],
  }));
}
```

Reply:
Updated here https://github.com/elastic/kibana/pull/187922/commits/269523601403ff39b9f1516ea2ca2a922374f759